modules/51DegreesRtdProvider.js

```diff
@@ -0,0 +1,255 @@
+const ORTB_DEVICE_TYPE = {
+  UNKNOWN: 0,
+  MOBILE_TABLET: 1,
+  PERSONAL_COMPUTER: 2,
+  CONNECTED_TV: 3,
+  PHONE: 4,
+  TABLET: 5,
+  CONNECTED_DEVICE: 6,
+  SET_TOP_BOX: 7,
+  OOH_DEVICE: 8
+};
+
+// Map of 51Degrees device types to ORTB device types. See
+// https://51degrees.com/developers/property-dictionary?item=Device%7CDevice
+// for available properties and values.
+const ORTB_DEVICE_TYPE_MAP = new Map([
+  ['Phone', ORTB_DEVICE_TYPE.PHONE],
+  ['Console', ORTB_DEVICE_TYPE.SET_TOP_BOX],
+  ['Desktop', ORTB_DEVICE_TYPE.PERSONAL_COMPUTER],
+  ['EReader', ORTB_DEVICE_TYPE.PERSONAL_COMPUTER],
+  ['IoT', ORTB_DEVICE_TYPE.CONNECTED_DEVICE],
+  ['Kiosk', ORTB_DEVICE_TYPE.OOH_DEVICE],
+  ['MediaHub', ORTB_DEVICE_TYPE.SET_TOP_BOX],
+  ['Mobile', ORTB_DEVICE_TYPE.MOBILE_TABLET],
+  ['Router', ORTB_DEVICE_TYPE.CONNECTED_DEVICE],
+  ['SmallScreen', ORTB_DEVICE_TYPE.CONNECTED_DEVICE],
+  ['SmartPhone', ORTB_DEVICE_TYPE.MOBILE_TABLET],
+  ['SmartSpeaker', ORTB_DEVICE_TYPE.CONNECTED_DEVICE],
+  ['SmartWatch', ORTB_DEVICE_TYPE.CONNECTED_DEVICE],
+  ['Tablet', ORTB_DEVICE_TYPE.TABLET],
+  ['Tv', ORTB_DEVICE_TYPE.CONNECTED_TV],
+  ['Vehicle Display', ORTB_DEVICE_TYPE.PERSONAL_COMPUTER]
+]);
+
+/**
+ * Extracts the parameters for 51Degrees RTD module from the config object passed at instantiation
+ * @param {Object} moduleConfig Configuration object of the 51Degrees RTD module
+ * @param {Object} reqBidsConfigObj Configuration object for the bidders, currently not used
+ */
+export const extractConfig = (moduleConfig, reqBidsConfigObj) => {
+  // Resource key
+  const resourceKey = deepAccess(moduleConfig, 'params.resourceKey');
+  // On-premise JS URL
+  const onPremiseJSUrl = deepAccess(moduleConfig, 'params.onPremiseJSUrl');
+
+  if (!resourceKey && !onPremiseJSUrl) {
+    throw new Error(LOG_PREFIX + ' Missing parameter resourceKey or onPremiseJSUrl in moduleConfig');
+  } else if (resourceKey && onPremiseJSUrl) {
+    throw new Error(LOG_PREFIX + ' Only one of resourceKey or onPremiseJSUrl should be provided in moduleConfig');
+  }
+  if (resourceKey === '<YOUR_RESOURCE_KEY>') {
+    throw new Error(LOG_PREFIX + ' replace <YOUR_RESOURCE_KEY> in configuration with a resource key obtained from https://configure.51degrees.com/tWrhNfY6');
+  }
+
+  return {resourceKey, onPremiseJSUrl};
+}
+
+/**
+ * Gets 51Degrees JS URL
+ * @param {Object} pathData API path data
+ * @param {string} [pathData.resourceKey] Resource key
+ * @param {string} [pathData.onPremiseJSUrl] On-premise JS URL
+ * @returns {string} 51Degrees JS URL
+ */
+export const get51DegreesJSURL = (pathData) => {
+  if (pathData.onPremiseJSUrl) {
+    return pathData.onPremiseJSUrl;
+  }
+  return `https://cloud.51degrees.com/api/v4/${pathData.resourceKey}.js`;
+}
+
+/**
+ * Injects 51Degrees script into the document head
+ * @param {string} url 51Degrees JS URL
+ * @returns {Promise} Promise that resolves when the script is loaded
+ */
+export const inject51DegreesScript = (url) => {
```

Comment:
You have to use our injection methods

Reply:
@patmmccann thanks for the comment.  Just wanted to clarify:
1. by "our injection methods" - do you mean `loadExternalScript` API from `adloader.js`?   
2. If so we'd need to add our module code into `_approvedLoadExternalJSList`, I assume it should also be part of this PR, right? 
3. by "another method" do you mean that the user may insert the `script` tag themselves at the point of embedding Prebid.js into the page when providing configuration, or is it something else?

Reply:
Yes, yes, yes